File: functions/ffmpeg/variants/go/2_vp5_adjust_2.go
package main

func vp5_adjust(v int32, t int32) int32 {
    s1 := v >> 31
    v = (v ^ s1) - s1
    flag := v < 2*t
    v *= int32(flag)
    v -= t
    s2 := v >> 31
    v = (v ^ s2) - s2
    return (t - v + s1) ^ s1
}


func main(){
    vp5_adjust(0,0)
}

File: functions/ffmpeg/variants/c/2_vp5_adjust_2.c
#include <stdint.h>

static int vp5_adjust(int v, int t)
{
    int s1 = v >> 31;
    v = (v ^ s1) - s1;
    int mask = (v < 2 * t) - 1;
    v = v & ~mask;
    v = v - t;
    int s2 = v >> 31;
    v = (v ^ s2) - s2;
    v = t - v;
    return (v ^ s1) + s1;
}


int main(){
    vp5_adjust(0,0);
}
